packages/blunderbuss/src/blunderbuss.ts

```diff
@@ -116,7 +137,17 @@ export = (app: Application) => {
         return;
       }
 
-      const assignee = randomFrom(assignees, issuePayload.user.login);
+      const labels = issue.labels?.map(l => l.name);
+      const possibleAssignees = assigneesForLabels(assignConfig, labels);
+      // If we didn't find a label based assignee, look for a general one.
+      if (possibleAssignees.length === 0) {
+        for (const a of assignConfig) {
```

Comment:
Rather than iterate through these twice (once here and once in `assignessForLabels`) would it make more sense to do both in a single pass, and only use/return regular assignees if there are no label specific assignees?

Reply:
Done. Needed an extra param to be able to use the same function to decide whether or not to process a labeled event.